src/Orchard.Web/Modules/Orchard.Taxonomies/Handlers/TermsPartHandler.cs

```diff
@@ -60,13 +60,22 @@ public class TermsPartHandler : ContentHandler {
             var queryHint = new QueryHints()
                 .ExpandRecords("ContentTypeRecord", "CommonPartRecord", "TermsPartRecord");
 
+            // Get TermRecordIds for each field before the delegate inside the for iterator.
+            // This avoids the TermsPart.Record lifetime scope to be disposed when executed, since ContentPart.Record is a LazyField.
+            var groupedRecordIds = part.Record.Terms
```

Comment:
> Could you check the SQL traces before and after the change? It is possible that now you are doing queries all the time instead of only when necessary, or even that you are loading records that would have never been before. Hence the reason of "lazy" loading.

Here's the test I did on the matter, after enabling NHibernate logging:
I created a new content type, adding a TaxonomyField to it. After that, I displayed the content item at front end in the following four scenarios:
1) Old loader, TaxonomyField displayed
2) Old loader, TaxonomyField hidden (via placement.info)
3) New loader, TaxonomyField displayed
4) New loader, TaonomyField hidden (via placement.info)

In each case the logger logged the same following queries:
- One query on Orchard_Taxonomies_TermsPartRecord, filtered by column "id" = the id of the displayed content item.
- One query on Orchard_Taxonomies_TermContentItem, filtered by column "TermsPartRecord_id" = the id of the displayed content item.
- One query on Orchard_Framework_ContentItemVersionRecord joined with other tables, the notable of them are Orchard_Framework_ContentItemRecord and Orchard_Taxonomies_TermsPartRecord, where ContentItemRecord.Id is the id of the selected term for the displayed content item.

This means that the new loader doesn't add (or remove) any query, but caches some info (the TermsPartRecord) to avoid accessing a disposed session or transaction later in the call.

The query to get the record is executed inside the StorageFilter for the repository (of TermsPartRecord), which is added here:
https://github.com/OrchardCMS/Orchard/blob/50d416c9f2b1ebecd56bfbb0c8933768493fc89f/src/Orchard.Web/Modules/Orchard.Taxonomies/Handlers/TermsPartHandler.cs#L34
The Record is loaded in any case here:
https://github.com/OrchardCMS/Orchard/blob/50d416c9f2b1ebecd56bfbb0c8933768493fc89f/src/Orchard/ContentManagement/Handlers/StorageFilter.cs#L30